### ISSUE TITLE
Path to macOS executable wrong in installation instructions

### ISSUE THREAD

**Issue by flipphillips** [kenkangxgwe/vscode-lsp-wl#1]
Should be `/Applications/Mathematica.app/Contents/MacOS/Wolfram`

I'd pull and change it myself, but I'm tired :)

**Comment by singularitti** [comment]
I am confused by this issue. On `README.md`, it says

> Wolfram Path: The path of the Wolfram executable. (NOT Mathematica or WolframKernel)

But on macOS, `/Applications/Mathematica.app/Contents/MacOS/Wolfram` is merely a link to `WolframKernel`
```shell
~ ❯❯❯ la /Applications/Mathematica.app/Contents/MacOS/Wolfram
lrwxr-xr-x  1 singularitti  admin    13B Apr  8 12:09 /Applications/Mathematica.app/Contents/MacOS/Wolfram -> WolframKernel
```

**Comment by kenkangxgwe** [comment]
Hi @singularitti, `Wolfram` and `WolframKernel` are different on Windows. What is the name of the non-CLI Kernel application, which looks like the image below, on MacOS?

![image](https://user-images.githubusercontent.com/8650023/59152433-bbb83700-8a11-11e9-8443-4a36d7e3b5c4.png)

**Comment by singularitti** [comment]
And also, it says

> For MaxOS users, the default path is `/Applications/Mathematica.app/Contents/wolfram`.

There is no `/Applications/Mathematica.app/Contents/wolfram`, the only one is `/Applications/Mathematica.app/Contents/Wolfram`. Though macOS does not seem to distinguish capitalized or not-capitalized word, it still confused me at the beginning. I was then trying to set
```json
"WolframLanguageServer.WolframPath": "/usr/local/bin/wolframscript",
```
because I cannot find `/Applications/Mathematica.app/Contents/Wolfram` through `ls`. I wish this doc could be more user-friendly. I could make a PR too.

**Comment by kenkangxgwe** [comment]
Yes, please create a PR. Sorry for any inconvenience since I do not use Mac.

**Comment by singularitti** [comment]
Hi, I guess what you are showing here is `WolframScript.app`. It is an Application sits aside the `Mathematica.app`.
![WolframScript.app](https://user-images.githubusercontent.com/25192197/59152463-67fa1d80-8a12-11e9-9055-2c970c7019ad.png)

This app on macOS has been linked to `/usr/local/bin/wolframscript`, as shown below
![/usr/local/bin/wolframscript](https://user-images.githubusercontent.com/25192197/59152489-cfb06880-8a12-11e9-907c-1ab769dc88f7.png)

But one needs to manually install `WolframScript.app`, i.e., it is optional to install. It is distributed together with `Mathematica.app`, though.

So if a user does not install `WolframScript.app`, he could only use the `wolframscript` the `Mathematica.app` provides. I bet they are the same. The `Mathematica.app`'s `wolframscript` is just not linked to `/usr/local/bin`. So under `/Applications/Mathematica.app/Contents/MacOS` directory, we have a bunch of executables
```shell
/Applications/Mathematica.app/Contents/MacOS ❯❯❯ ls
MathKernel        MathematicaScript WolframKernel     wolframscript
Mathematica       MathematicaServer wolfram
```

![ls](https://user-images.githubusercontent.com/25192197/59152454-50bb3000-8a12-11e9-833d-5e5fdcfd6677.png)
As shown above, `MathKernel`, `MathematicaScript` and `wolfram` are just symolic links, which I am not going to talk about. `Mathematica`, of course, is `Mathematica`. `WolframKernel` and `wolframscript`, they print the same banners, as shown above, so I guess they are the same.

```shell
/Applications/Mathematica.app/Contents/MacOS ❯❯❯ ./wolframscript
Wolfram Language 12.0.0 Engine for Mac OS X x86 (64-bit)
Copyright 1988-2019 Wolfram Research, Inc.

In[1]:=                  
```

In conclusion, we could either let it be `/Applications/Mathematica.app/Contents/MacOS/WolframKernel`, or `/Applications/Mathematica.app/Contents/MacOS/wolframscript`, or their symbolic links.

I hope I have clarified things.

**Comment by singularitti** [comment]
Wait, there is `/Applications/Mathematica.app/Contents/wolfram`. I am sorry that I made a mistake, but I hope I made it clear what macOS has.

**Comment by kenkangxgwe** [comment]
Yeah, I see what you meant. For this specific plugin, it requires the `wolfram` to pass the `-script` switch.
https://github.com/kenkangxgwe/vscode-lsp-wl/blob/872e57f75b41e08dfef72470ad0325a3a97696c5/src/extension.ts#L28

I did not use `wolframscript` because nodejs cannot communicate with it for some reason :(.

**Comment by singularitti** [comment]
Great, I guess
```shell
/Applications/Mathematica.app/Contents/MacOS/wolframscript
/Applications/Mathematica.app/Contents/MacOS/WolframKernel
/Applications/Mathematica.app/Contents/MacOS/wolfram
/Applications/Mathematica.app/Contents/MacOS/MathKernel
```
are equivalent. I am not sure why Mathematica has so much equivalent executables, maybe it is for compatibility. But anyway, it does not need a PR. :)
